### PR TITLE
Add fine functionality

### LINKED ISSUE #81: Implement Fines
For the new re-work, it is critical that fines are implemented before this is going in production. Features to implement are:

- [x] Allow users to see how much they have in fines.
- [x] Allow the BAC PM to Notify people eligible for fines
- [x] Allow the BAC PM to easily handout fines, based on two reference dates
- [x] Allow the BAC PM to delete fines
- [x] (Optional) Allow the BAC PM to have an overview of how much money is open in outstanding fines.
- [x] Allow people to see fines in their transaction overview.

### LINKED ISSUE #111: Quick overview shows internal ID on reload
When reloading the page, the quick overview on mobile shows the internal SudoSOS id

### LINKED ISSUE #112: Transaction table is one month behind
Just an off by one error

### PR DESCRIPTION
**Pull request by CodeNamedRobin**
Fine overview is on route /fine
Users can now see how much they have in fines.

Also fixes #111,
And fixes #112,
Also fixes #81,

### CODE REVIEW

**Comment by rinkp** [comment]
What do you need from me?
________________________________

**Comment by SuperVK** [comment]
Is there a reason it is hidden in /fine, and not just a RBAC hidden menu button only for BAC PM?

**Comment by SuperVK** [comment]
The handout fine button does not give any feedback, maybe a success toast? Just fined myself 5 times wondering why it wasn't working lol.

**Comment by SuperVK** [comment]
image
The info modal of the fine seems to be missing, maybe we can remove this modal all together? Or a short info about fines and how they work?

**Review by SuperVK — APPROVED** [review]
LGTM

File: src/views/FineView.vue [hunk 1]
```diff
@@ -122,32 +191,77 @@ const handlePickedDates = handleSubmit(async (values) => {
...
 onMounted(async () => {
   await userStore.fetchUsers(apiService);
-  console.log(userStore.users);
+  fineHandoutEvents.value = await fetchAllPages<FineHandoutEventResponse>(
+    0,
+    Number.MAX_SAFE_INTEGER,
+    // @ts-ignore
+    (take, skip) => apiService.debtor.returnAllFineHandoutEvents(take, skip)
+  );
+  console.log(fineHandoutEvents.value);
```
Comment (tomudding) [inline src/views/FineView.vue:221]: 👀

**Pull request merged**